app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Nutriments.java

```diff
@@ -197,11 +197,16 @@
     private boolean containsMinerals;
 
     public Nutriment get(String nutrimentName){
-        if (!additionalProperties.containsKey(nutrimentName)) {
+        if (nutrimentName.isEmpty() || !additionalProperties.containsKey(nutrimentName)) {
             return null;
         }
 
-        return new Nutriment(additionalProperties.get(nutrimentName).toString(), get100g(nutrimentName), getServing(nutrimentName), getUnit(nutrimentName));
+        try{
+            return new Nutriment(additionalProperties.get(nutrimentName).toString(), get100g(nutrimentName), getServing(nutrimentName), getUnit(nutrimentName));
+        }catch (NullPointerException e){
+        // In case one of the getters was unable to get data as string
```

Comment:
You should print something to the log here for troubleshooting issues. Even `e.printStackTrace()` would be fine.

Reply:
Good idea 👍